src/__tests__/pages/api/auth/reset-password.test.ts

```diff
@@ -0,0 +1,299 @@
+import dotenv from 'dotenv';
+dotenv.config();
+
+import {
+  mockUserData,
+  mockPassword,
+  mockEmail
+} from '../../../../__mocks__/pages/api/auth';
+import signUpHandler from '../../../../pages/api/auth/signup';
+import resetPasswordHandler from '../../../../pages/api/auth/reset-password';
+import { PrismaClient } from '@prisma/client';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { hashFunction } from '../../../../utils/password';
+import { resetPasswordProps } from '../../../../types/api/auth/reset-password';
+
+const prisma = new PrismaClient();
+
+let email: string, password: string;
+
+beforeEach(async () => {
+  const userData = mockUserData();
+  email = userData.email;
+  password = userData.password;
+
+  const req = {
+    body: {
+      email,
+      password
+    }
+  } as NextApiRequest;
+
+  const json = jest.fn();
+
+  const status = jest.fn(() => {
+    return { json };
+  });
+
+  const res = {
+    status
+  } as unknown as NextApiResponse;
```

Comment:
as unknown?

Reply:
I got error by typescript when I tried without unknown,
```txt
Type '{ status: Mock<{ json: Mock<any, any>; }, []>; }' is missing the following properties from type 'ServerResponse': statusCode, statusMessage, assignSocket, detachSocket, and 54 more.
```
this indicates that a lot of properties of type `ServerResponse` are not defined, and it's efficient to not define them. Adding `unknown` is like telling it's an unstrict type `ServerResponse`